apps/dashboard/src/components/workflow-list.tsx

```diff
@@ -83,46 +143,117 @@ export const WorkflowList = () => {
           <TableRow>
             <TableCell colSpan={5}>
               <div className="flex items-center justify-between">
-                <p className="text-foreground-600 text-sm font-normal">Page X of Y</p>
-                <Pagination>
-                  <PaginationContent>
-                    <PaginationItem>
-                      <PaginationStart href="#" />
-                    </PaginationItem>
-                    <PaginationItem>
-                      <PaginationPrevious href="#" />
-                    </PaginationItem>
-                    <PaginationItem>
-                      <PaginationLink href="#">1</PaginationLink>
-                    </PaginationItem>
-                    <PaginationItem>
-                      <PaginationLink href="#">2</PaginationLink>
-                    </PaginationItem>
-                    <PaginationItem>
-                      <PaginationLink href="#">3</PaginationLink>
-                    </PaginationItem>
-                    <PaginationItem>
-                      <PaginationEllipsis />
-                    </PaginationItem>
-                    <PaginationItem>
-                      <PaginationLink href="#">15</PaginationLink>
-                    </PaginationItem>
-                    <PaginationItem>
-                      <PaginationNext href="#" />
-                    </PaginationItem>
-                    <PaginationItem>
-                      <PaginationEnd href="#" />
-                    </PaginationItem>
-                  </PaginationContent>
-                </Pagination>
-                <Select>
+                {workflowsQuery.data ? (
+                  <span className="text-foreground-600 block text-sm font-normal">
+                    Page {currentPage} of {Math.ceil(workflowsQuery.data.totalCount / limit)}
+                  </span>
+                ) : (
+                  <Skeleton className="h-5 w-[20ch]" />
+                )}
+                {workflowsQuery.data ? (
+                  <Pagination>
+                    <PaginationContent>
+                      <PaginationItem>
+                        <PaginationStart href="#" onClick={() => setOffset(0)} />
+                      </PaginationItem>
+                      <PaginationItem>
+                        <PaginationPrevious href="#" onClick={() => setOffset(Math.max(0, offset - limit))} />
+                      </PaginationItem>
+                      {(() => {
+                        const currentPage = Math.floor(offset / limit) + 1;
+                        const totalPages = Math.ceil(workflowsQuery.data.totalCount / limit);
+                        const startPage = Math.max(1, currentPage - 2);
+                        const endPage = Math.min(totalPages, currentPage + 2);
+
+                        const pageItems = [];
```

Comment:
When I have just only one page the pagination buttons are not disabled:

https://github.com/user-attachments/assets/900f595b-3184-4451-b4f1-dd25c1d6c644